src/proxy.cc

```diff
@@ -243,27 +243,30 @@ ncclResult_t getOpIndex(struct ncclProxyArgs* op, struct ncclProxyProgressState*
 }
 
 ncclResult_t printProxyOp(struct ncclProxyArgs* op, int poolIndex, int opIndex) {
-  printf("[%d-%d|%ld| %s", poolIndex, opIndex, op->opCount, op->pattern == ncclPatternSend ? "Send" : op->pattern == ncclPatternRecv ? "Recv" : "Coll");
+  int peer = op->send ? op->nextRank : op->prevRank;
+  int isColl = (op->pattern != ncclPatternSend && op->pattern != ncclPatternRecv);
+  printf("[%d-%d|%ld| %s", poolIndex, opIndex, op->opCount, isColl ? "Coll->" : "");
+  printf("%s", op->send ? "Send" : "Recv");
   for (int s=0; s<op->nsubs; s++) {
     struct ncclProxySubArgs* sub = op->subs+s;
     if (op->state == ncclProxyOpProgress) {
       char status = ' ';
-      if (op->pattern == ncclPatternRecv) {
+      if (! op->send) {
         if (sub->posted < sub->nsteps && sub->posted < sub->done + NCCL_STEPS) status = 'I'; // Init
         else if (sub->received < sub->posted) status = 'R'; // Receiving
         else if (sub->received < sub->transmitted) status = 'R'; // Receiving
         else if (sub->transmitted < sub->received) status = 'F'; // Flushing
         else if (sub->done < sub->transmitted) status = 'G'; // Waiting on GPU
         else status = 'D'; // Done
-      } else if (op->pattern == ncclPatternSend) {
+      } else {
```

Comment:
There are many "patterns". Why remove the original checking?

Reply:
ncclPatternSend/Recv is used in only ncclSend/ncclRecv. In this PR, we want to see all the send/recv inside a collective operation (such as allreduce) and this pattern check prevents it.

Reply:
I am not sure I understand the reason. If this PR is about dump proxy information, it should not change proxy code handling itself. Unless original proxy code was wrong here? We can add proxy dump at all if/else if/else conditions without changing logic of proxy code.